wallet/src/wallet/mod.rs

```diff
@@ -163,11 +180,30 @@ impl<B: storage::Backend> Wallet<B> {
 
         db_tx.close();
 
+        let first: &Account = accounts.values().next().ok_or(WalletError::WalletNotInitialized)?;
+        let last = accounts.values().last().ok_or(WalletError::WalletNotInitialized)?;
+
+        let unsynced_account = if first.best_block() != last.best_block() {
+            accounts.pop_last()
+        } else {
+            None
+        };
+
+        // sanity check all other accounts have the same best block
+        ensure!(
+            accounts
+                .values()
+                .tuple_windows()
+                .all(|(acc1, acc2)| acc1.best_block() == acc2.best_block()),
+            WalletError::WalletNotInitialized
```

Comment:
Can't we instead fix this problem? What's the benefit of being so harsh and fail loading the wallet?